sentry/src/main/java/io/sentry/transport/HttpConnection.java

```diff
@@ -27,37 +24,21 @@
 import javax.net.ssl.HostnameVerifier;
 import javax.net.ssl.HttpsURLConnection;
 import javax.net.ssl.SSLSocketFactory;
-import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 import org.jetbrains.annotations.TestOnly;
 
-/**
- * An implementation of the {@link ITransport} interface that sends the events to the Sentry server
- * over HTTP(S).
- */
 @Open
-@ApiStatus.NonExtendable // only not final because of testing
-@ApiStatus.Internal
-public class HttpTransport implements ITransport {
+class HttpConnection {
```

Comment:
is the `HttpConnection` reusable for the NIO implementation as well? just trying to understand why we have the BlockingHttpTransport + HttpConnection, I do get their difference, but what would be the advantage of it right now?

Reply:
Just to split the responsibilities - and I agree that the split is not perfect. I tried to create decorators like `EnvelopeCachingHttpTransport` and `RateLimitingHttpTransport` but since `send` returns `void` it's also not possible. 

`HttpConnection` is package private, to give us flexibility of refactoring later without breaking users' code.

Reply:
sounds good to me